### PR TITLE
fix JsonFraming to accept multiple top-level arrays (#26098)

### PR DESCRIPTION
**Pull request by shkoder**
ref #26098

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/JsonFramingSpec.scala [hunk 1]
```diff
@@ -126,7 +153,7 @@ class JsonFramingSpec extends AkkaSpec {
       val input = TestPublisher.probe[ByteString]()
       val output = TestSubscriber.probe[String]()
 
-      val result = Source.fromPublisher(input)
+      Source.fromPublisher(input)
```
Comment (shkoder) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/JsonFramingSpec.scala]: removes unused warning

File: akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala [hunk 2]
```diff
@@ -56,7 +56,6 @@ import scala.annotation.switch
   private var trimFront = 0 // number of chars to drop from the front of the bytestring before emitting (skip whitespace etc)
   private var depth = 0 // counter of object-nesting depth, once hits 0 an object should be emitted
 
-  private var charsInObject = 0
```
Comment (shkoder) [inline akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala]: was only updated (removed line 143 below), never queried

File: akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala [hunk 3]
```diff
@@ -118,7 +117,8 @@ import scala.annotation.switch
       trimFront += 1
     } else if (input == SquareBraceEnd && outsideObject) {
       // outer array completed!
-      pos = -1
+      pos += 1
```
Comment (shkoder) [inline akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala]: the fix

**Comment by johanandren** [comment]
Note that there is already quite a big rework of the JsonFraming in #25674 so please check if that solves this issue as well

**Comment by johanandren** [comment]
(We could probably merge this before that though, seems simple enough)

**Comment by raboof** [comment]
OK TO TEST

**Comment by raboof** [comment]
PLS BUILD

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by helena** [comment]
Restarted all checks to see where this is.

**Comment by helena** [comment]
@raboof jcenter strikes again https://travis-ci.org/akka/akka/jobs/570700857

**Comment by jrudolph** [comment]
> jcenter for Scala 2.12.8 strikes again [travis-ci.org/akka/akka/jobs/570700857](https://travis-ci.org/akka/akka/jobs/570700857)
> com.lightbend.akka#paradox-theme-akka;0.23: not found
> 2.13 is fine.

Restarted the failing job.

**Review by helena — APPROVED** [review]
LGTM, in light of this going before https://github.com/akka/akka/pull/25674.

**Comment by shkoder** [comment]
created #27780 to backport to 2.5

**Pull request merged**